### PR TITLE
add RFC template

### PR DESCRIPTION
**Pull request by drprajap**
Please review these guidelines to help with the review process:
- [ ] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [ ] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: docs/rfcs/template.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# IMEX Design Document (RFC)
+
+## Summary
+Short description of proposed solution.
+
+## Motivation
+Explain motivation for the proposed idea. 
+If you are proposing a new Dialect, it needs to address below criteria:
+- The new dialect needs to introduce Intel unique features which we don’t think upstream dialect has interests to cover.
```
Comment (Jianhui-Li) [inline docs/rfcs/template.md:9]: Intel unique features => unique features.

File: docs/rfcs/template.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# IMEX Design Document (RFC)
+
+## Summary
+Short description of proposed solution.
+
+## Motivation
+Explain motivation for the proposed idea. 
+If you are proposing a new Dialect, it needs to address below criteria:
```
Comment (Jianhui-Li) [inline docs/rfcs/template.md:8]: If you are proposing an optimization pass, please explain the applicable scenario and expected performance gain.

File: docs/rfcs/template.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- The new dialect needs to introduce Intel unique features which we don’t think upstream dialect has interests to cover.
+- The new dialect should facilitate certain compile-time optimizations which otherwise cannot or very hard to achieve. In other words, there should be compile-time optimizations which requires the computation graph to be represented using the new dialects. If there is no compile-time optimization required, the functions could be represented  as a standardized MLIR module to facilitate code reuse.	
+
+## Proposal
+A Full and detailed description of proposal.
+
+## Alternative
+Proposal needs to mention any alternative that were being considered with pros and cons. Instead of creating a new dialect to introduce intel specific ops, consider alternative  to add specific ops in the existing upstream dialects as extension.
+
```
Comment (Jianhui-Li) [inline docs/rfcs/template.md:17]: For optimization passes, describe alternative optimization approaches.

**Review by Jianhui-Li — APPROVED** [review]

File: docs/rfcs/template.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# IMEX Design Document (RFC)
+
+## Summary
+Short description of proposed solution.
+
+## Motivation
+Explain motivation for the proposed idea.
+If you are proposing an optimization pass, please explain the applicable scenario and expected performance gain. It needs to address below criteria:
+- The new dialect needs to introduce unique features which we don’t think upstream dialect has interests to cover.
+- The new dialect should facilitate certain compile-time optimizations which otherwise cannot or very hard to achieve. In other words, there should be compile-time optimizations which requires the computation graph to be represented using the new dialects. If there is no compile-time optimization required, the functions could be represented  as a standardized MLIR module to facilitate code reuse.
```
Comment (fschlimb) [inline docs/rfcs/template.md:10]: I am confused: This defines criteria for _optimizations passes_ but the criteria talk about characteristics of the _dialects_. I am not sure if the distinction between passes and dialects is intended, but if it is then this is unclear.

@Jianhui-Li @drprajap What is a "standard MLIR module"?

Comment (drprajap) [reply docs/rfcs/template.md:10]: Above criteria should be applicable to both pass and dialect. there is no distinction intended here. Standard MLIR module could be referred to as high level function that utilizes existing/upstream dialects/passes but provides certain flow of operations enclosed within a function. One example that I can think of is something similar as described [here](https://mlir.llvm.org/docs/LangRef/#high-level-structure)

**Review by fschlimb — APPROVED** [review]

**Pull request merged**